### DISCUSSION TITLE
Problems with versions 0.5.0 and 0.7.1

### DISCUSSION THREAD

**Discussion by SergioStekolschik** [provectus/kafka-ui#4010] [Q&A] [answered]
Sorry, to continue with the question #3997 (registrations cannot be null or empty)

Thank you very much for your answers, unfortunately I did not find the solution to my problems in the recommended links and the Q&A is closed.

I summarize:

Topic 1- Secure connection to MSK:

KAFKA_CLUSTERS_0_NAME=MSK-CLT-INTEGRATION-INTEGRATIONBUS
KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS=xxxxx:9096
KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL=SASL_SSL
KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=SCRAM-SHA-512
KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG=org.apache.kafka.common.security.scram.ScramLoginModule required username="xxxxx" password="xxxxx";
KAFKA_CLUSTERS_0_PROPERTIES_PROTOCOL=SASL
KAFKA_CLUSTERS_0_ZOOKEEPER=xxxxx:9096
KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_LOCATION=/etc/kafka/secrets/kafka.truststore.jks
KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_PASSWORD=xxxxx
KAFKA_CLUSTERS_0_READONLY=true

Version 0.5.0: It works perfectly.
Version 0.7.1: I get the errors

Failed to collect cluster MSK-CLT-INTEGRATION-INTEGRATIONBUS info java.lang.IllegalStateException: Error while creating AdminClient for Cluster MSK-CLT-INTEGRATION-INTEGRATIONBUS

Caused by: org.apache.kafka.common.KafkaException: org.apache.kafka.common.KafkaException: Failed to load SSL keystore /etc/kafka/secrets/kafka.truststore.jks of type JKS

Topic 2- Use of OAuth2:

AUTH_TYPE=OAUTH2
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENT_NAME=Kafka-UI
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENTID=xxxxx
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENTSECRET=xxxxx
AUTH_OAUTH2_CLIENT_AZUREAD_SCOPE=https://graph.microsoft.com/User.Read
AUTH_OAUTH2_CLIENT_AZUREAD_PROVIDER=AZURE
AUTH_OAUTH2_CLIENT_AZUREAD_ISSUER_URI=https://login.microsoftonline.com/xxxx/v2.0
AUTH_OAUTH2_CLIENT_AZUREAD_REDIRECT_URI=https://127.0.0.1:8080/login/code/AZURE
AUTH_OAUTH2_CLIENT_AZUREAD_AUTHORIZATION_GRANT_TYPE=authorization_code

Version 0.5.0: I get the following errors:

Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'OAuthLogoutSuccessHandler'

Error creating bean with name 'defaultOidcLogoutHandler' defined in class path resource

Error creating bean with name 'clientRegistrationRepository' defined in class path resource

Failed to instantiate [org.springframework.security.oauth2.client.registration.InMemoryReactiveClientRegistrationRepository]: Factory method 'clientRegistrationRepository' threw exception; nested exception is java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because the return value of "com.provectus.kafka.ui.config.auth.OAuthProperties$OAuth2Provider.getCustomParams()" is null
at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
... 55 common frames omitted

Version 0.7.1: I get the following errors:

In the web browser:

Secure Connection Failed
An error occurred during a connection to 127.0.0.1:8080. SSL received a record that exceeded the maximum permissible length.
Error code: SSL_ERROR_RX_RECORD_TOO_LONG

Regards,

Sergio

**Comment by Haarolean** [comment, marked as answer]
For the first issue, refer to 0.6.0 changelog's breaking changes section: [here](https://github.com/provectus/kafka-ui/releases/tag/v0.6.0)

For the second one, you're trying to access localhost:8080 with SSL enabled (HTTPS in your url) which doesn't have SSL enabled on this endpoint. Either use HTTP or put your kafka-ui behind the reverse proxy (or enable one right in k-ui (unrecommended): [here](https://github.com/provectus/kafka-ui/discussions/2384))

**Reply by SergioStekolschik** [reply]
Hi Haarolean,

Thank you very much for your reply.
First issue:

For version (0.7.1), I made the change of:

KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_LOCATION=/etc/kafka/secrets/kafka.truststore.jks
KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_PASSWORD=xxxxx

By:

KAFKA_CLUSTERS_0_SSL_TRUSTSTORELOCATION=/etc/kafka/secrets/kafka.truststore.jks
KAFKA_CLUSTERS_0_SSL_TRUSTSTOREPASSWORD=xxxxx

The password with and without quotes.
I still get the errors:

* Failed to collect cluster MSK-CLT-INTEGRATION-INTEGRATIONBUS info java.lang.IllegalStateException: Error while creating AdminClient for Cluster MSK-CLT-INTEGRATION-INTEGRATIONBUS

* Caused by: org.apache.kafka.common.KafkaException: org.apache.kafka.common.KafkaException: Failed to load SSL keystore /etc/kafka/secrets/kafka.truststore.jks of type JKS

I attach images:

![ImageCertificateInContainer](https://github.com/provectus/kafka-ui/assets/20090377/e1773eda-1b12-4caf-8cd3-722094023164)
![ImageError1](https://github.com/provectus/kafka-ui/assets/20090377/a25d3338-c8b4-45c2-b043-78b3ebc82932)
![ImageError2](https://github.com/provectus/kafka-ui/assets/20090377/1e0b10f9-ca9a-41f9-81f1-69d42fe0c6ea)

Second issue:

I could not finish to configure NGINX correctly.

Regards,

Sergio

**Reply by Haarolean** [reply]
Please read the stacktrace carefully, it clearly indicates the AccessDeniedException. Check your file system permissions.

**Reply by SergioStekolschik** [reply]
Hi Haarolean,

Thank you very much for your reply.
* First issue:

For version (0.7.1), 

The problem was that I did not delete the following settings:

KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_LOCATION=/etc/kafka/secrets/kafka.truststore.jks
KAFKA_CLUSTERS_0_SSL_TRUSTSTORE_PASSWORD=xxxxx

Now the Kafka-UI container connects to AWS MSK without any problems.

* Second issue:

Finally I deployed the container on AWS Fargate with the Application Load Balancers with this configuration:

AUTH_TYPE=OAUTH2
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENT_NAME=Kafka-UI
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENTID=xxxxx
AUTH_OAUTH2_CLIENT_AZUREAD_CLIENTSECRET=xxxxx
AUTH_OAUTH2_CLIENT_AZUREAD_SCOPE=https://graph.microsoft.com/User.Read
AUTH_OAUTH2_CLIENT_AZUREAD_PROVIDER=AZURE
AUTH_OAUTH2_CLIENT_AZUREAD_ISSUER_URI=https://login.microsoftonline.com/xxxx/v2.0
AUTH_OAUTH2_CLIENT_AZUREAD_REDIRECT_URI=https://xxxxxxxx:8080/login/code/AZURE
AUTH_OAUTH2_CLIENT_AZUREAD_AUTHORIZATION_GRANT_TYPE=authorization_code

I login to Azure AD and when it comes back I receive the folling error:

![Image333](https://github.com/provectus/kafka-ui/assets/20090377/361756ef-b3da-4ab7-b1bc-eba7cf1f0f27)

Few seconds later the error change to:

![Image444](https://github.com/provectus/kafka-ui/assets/20090377/e2a0e8f1-e4f5-4609-8e33-71d9368a5d6a)

What could be the cause of the error?

Thanks in advance.
Regards,

Sergio
